### PR TITLE
Keep source file structure

### PR DESCRIPTION
**Pull request by jingsam**
Suppose we have two files:
```
/path/to/a.js
/other/path/to/a.js
```

In https://github.com/paazmaya/shuji/blob/master/index.js#L36
```
map[path.basename(source)] = contents;
```
Unfortunately, `/path/to/a.js` was overwrite.

This PR keep the file path and write to the directory which relative to the output_dir

### CODE REVIEW

**Comment by paazmaya** [comment]
Hi, thank you for the PR.
Looks like tests are failing.

How about when the source file path points to a location which does not exists, or is outside the current scope of the file tree?
I would not want the files written in surprisingly random places, but always under a certain folder.

Any ideas for that?

**Comment by evolutionxbox** [comment]
@paazmaya are you still wanting this pull request to be merged?

I would love it if it could be! ❤️

**Comment by paazmaya** [comment]
@evolutionxbox sure, but my comments needs to be addressed somehow first...

**Comment by jingsam** [comment]
@paazmaya Yes, you are right. We should make the outputdir as the root directory and all sources relative to the root. But currently I am a little busy to finish this work.  Maybe later I could done it

File: package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "shuji",
```
Comment (paazmaya) [inline package.json]: Sorry, not going to...

**Pull request closed without merging**